File: MyDent.Services/Database/Payment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using MyDent.Model.Enums;

namespace MyDent.Services.Database
{
    public class Payment
    {
        [Key]
        public int Id { get; set; }

        public int AppointmentId { get; set; }

        [ForeignKey("AppointmentId")]
        public Appointment Appointment { get; set; } = null!;

        // Actually-charged amount — server owns pricing, this is never recalculated
        // from current DentalService.Price for refund purposes.
        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal Amount { get; set; }

        [Required]
        public PaymentStatus Status { get; set; } = PaymentStatus.Pending;

        [MaxLength(200)]
        public string? ProviderTransactionId { get; set; }

        public DateTime? PaidAt { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? RefundedAmount { get; set; }

        public DateTime? RefundedAt { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: UI/mydent_mobile/lib/models/payment.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment.g.dart';

@JsonSerializable()
class Payment {
  final int? id;
  final int? appointmentId;
  final String? patientName;
  final String? doctorName;
  final double? amount;

  // Raw int matching PaymentStatus (see enums.dart).
  final int? status;
  final String? providerTransactionId;
  final DateTime? paidAt;
  final double? refundedAmount;
  final DateTime? refundedAt;
  final DateTime? createdAt;

  Payment({
    this.id,
    this.appointmentId,
    this.patientName,
    this.doctorName,
    this.amount,
    this.status,
    this.providerTransactionId,
    this.paidAt,
    this.refundedAmount,
    this.refundedAt,
    this.createdAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) =>
      _$PaymentFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentToJson(this);
}
